src/segment_owm.go

```diff
@@ -61,22 +63,57 @@ func (d *owm) string() string {
 	return text
 }
 
-func (d *owm) setStatus() error {
+func (d *owm) getResult() (*OWMDataResponse, error) {
 	apikey := d.props.getString(APIKEY, ".")
 	location := d.props.getString(LOCATION, "De Bilt,NL")
 	units := d.props.getString(UNITS, "standard")
 	timeout := d.props.getInt(HTTPTimeout, DefaultHTTPTimeout)
+	cachetimeout := int64(d.props.getInt(CACHETIMEOUT, 10))
+
+	d.url = fmt.Sprintf("http://api.openweathermap.org/data/2.5/weather?q=%s&units=%s&appid=%s", location, units, apikey)
+
+	if cachetimeout > 0 {
+		// check if data stored in cache
+		val, found := d.env.cache().get("owm_response")
+		// we got something from te cache
+		if found {
```

Comment:
Do an early return with `!found`

Reply:
in that case, if not found in the cache we continue the flow and call the service. If found we exit. Does an early return with !found apply here?